source/common/stats/stat_data_allocator_impl.h

```diff
@@ -87,6 +88,10 @@ template <class StatData> class CounterImpl : public Counter, public MetricImpl
   void reset() override { data_.value_ = 0; }
   bool used() const override { return data_.flags_ & Flags::Used; }
   uint64_t value() const override { return data_.value_; }
+  void stealthyAdd(uint64_t amount) override {
```

Comment:
I see what you did here, but I don't think this is entirely correct. Meaning, for low rate events we might lose some during the restart. I think we really do need to send over the deltas. Basically here is what we are looking for:
1) Child tells parent to stop flushing stats
2) Child asks parent for stats, during this, latch, and return delta.
3) Repeat 2 until we terminate the parent.

Reply:
Ahhhh, yes, I see: parent exports 100, 1 gets added, parent sends the value to child for the first time, child exports 101 but a delta of 0. Now that I'm thinking about it again, I don't think the extra complexity should be all that much. (Assuming it is the case that we can be sure that no other code will be calling latch() once the hot restart has begun, which appears true, although I'm uncertain about [load_stats_reporter.cc](https://github.com/envoyproxy/envoy/blob/814dd9facff27391a42744309f7765dc94971e40/source/common/upstream/load_stats_reporter.cc#L67).

Reply:
Done.